### ISSUE TITLE
Namespace issue with IE11

### ISSUE THREAD

**Issue by shaipan** [gwtproject/gwt#9428]
xmlns:NS1="" NS1: is getting appended in the request body in IE11.The similar issue was discussed in GWT-Openlayers
http://osgeo-org.1560.x6.nabble.com/WFS-and-IE-11-td5090636.html#a5094634

**Comment by tbroyer** [comment]
Can you please provide a repro case? Please note that the `com.google.gwt.xml.XML` module does **not** support *creating* elements and attributes using XML Namespaces.

**Comment by shaipan** [comment]
@tbroyer: I have following code to create the XML document.which works fine with all the browser except IE11.
		// Create document to save
		Document docToSave = XMLParser.createDocument();

		// Root element 
		Element rootElement = docToSave.createElement("gpmaint:maintenance");
		rootElement.setAttribute("xmlns:gpmaint","urn:vfi-sapphire:np.gpmaint.2004-08-03");		
		docToSave.appendChild(rootElement);

**Comment by tbroyer** [comment]
I can reproduce in IE11 with the following JavaScript code:
```js
var doc = document.implementation.createDocument("", "", null);
var elt = doc.createElement("foo:bar");
doc.appendChild(elt);
elt.setAttribute("xmlns:foo", "http://example.com");
elt.setAttribute("xmlns:baz", "http://example.net");
elt.setAttribute("baz:quux", "value");
window.alert(new XMLSerializer().serializeToString(elt));
```

(actually, even without the `xmlns:*` attributes it produces an `xmlns:NS1="" NS1:baz:quux="value"`)

So this is not a GWT bug, and given that this is a misuse of the API (one should create "namespaced elements" and "namespaced attributes" using `createElementNS` and `setAttributeNS`; for which GWT proper doesn't expose any built-in API, but that's another matter), we're not going to add any kind of workaround; and I'm closing this as _invalid_ rather than a _browser quirk._

FWIW, the correct way to build the above document would be (still in JavaScript):
```js
var doc = document.implementation.createDocument("", "", null);
var elt = doc.createElementNS("http://example.com", "foo:bar");
doc.appendChild(elt);
elt.setAttributeNS("http://example.net", "baz:quux", "value");
window.alert(new XMLSerializer().serializeToString(elt));
```
which in IE11 then gives `<foo:bar xmlns:foo="http://example.com" xmlns:baz="http://example.net" baz:quux="value" />`.

**Comment by shaipan** [comment]
@tbroyer We only write Java Code and not the JavaScript .GWT generates the javascript script for us.How to achieve this functionality using java code. Since there is no such api Element#setAttributeNS(......)

**Comment by tbroyer** [comment]
Either use JsInterop (if you're already on 2.8'ish), or JSNI (and then "write JavaScript in your Java code"). You could also use Elemental if you don't mind changing things again for Elemental 2 and GWT 2.8. There may be third-party libraries that give you access to these APIs too.

With Elemental, that'd read:
```java
Document doc = Browser.getDocument().getImplementation.createDocument("", "", null);
Element elt = doc.createElementNS("http://example.com", "foo:bar");
doc.appendChild(elt);
elt.setAttributeNS("http://example.net", "baz:quux", "value");
String str = Browser.getWindow().newXMLSerializer().serializeToString(doc);
```

**Comment by shaipan** [comment]
@tbroyer We are not GWT 2.5 and we are not allowed to make any javascript file changes directly .Let us know if you have any java workaround for this GWT version.

**Comment by tbroyer** [comment]
Elemental is available since GWT 2.5: https://search.maven.org/#search%7Cgav%7C1%7Cg%3A%22com.google.gwt%22%20AND%20a%3A%22gwt-elemental%22
…but GWT 2.5 is 4 years old already (one year earlier than IE11!)
